include/deal.II/grid/grid_tools.h

```diff
@@ -756,6 +756,88 @@ namespace GridTools
   std::vector<typename MeshType::active_cell_iterator>
   compute_ghost_cell_halo_layer (const MeshType &mesh);
 
+  /**
+   * Extract and return the active cell within a @p skin_thickness
+   * around a subdomain (set of active cells) in the @p mesh
+   * (i.e. those that share a common set of vertices with the subdomain but
+   * are not a part of it). Here, the "subdomain" consists of exactly all of
+   * those cells for which the @p predicate returns @p true.
+   *
+   * The function first computes interior active, boundary cells of the
+   * subdomain that conform to the predicate.
+   * Using compute_bounding_box, a bounding box is
+   * computed for this subdomain and extended by @p skin_thickness. All the
+   * active cells that are outside the extended bounding box are ignored.
+   * The cells that are inside the extended bounding box are then checked for
+   * their proximity to the interior subdomain boundary cells.
+   *
```

Comment:
You need to add another paragraph, something along the lines:

> This implies checking the distance between a pair of arbitrarily oriented cells, which is not trivial in general.
To simplify this, the algorithm checks the distance between the two small spheres that contain each cell. 
This will definitely result in slightly more cells being marked but also greatly simplifies the arithmetic complexity of the algorithm.